Frontend/Gmfctn/src/app/core/services/achievement-service/achievement-service.service.ts

```diff
@@ -0,0 +1,54 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, EMPTY, Observable, of } from 'rxjs';
+import { catchError, count, map, tap } from 'rxjs/operators';
+import { Achievement } from 'src/app/shared/models/achievement';
+import { apiUrl } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AchievementServiceService implements OnDestroy {
+
+  achievements$: BehaviorSubject<Achievement[]> = new BehaviorSubject(null as unknown as Achievement[]);
```

Comment:
I'd surggest to type this subject as BehaviorSubject<Achievement[] | null>. In this case you'll be able to get rid of this casting null to Achievement[] and will be better informed of what's inside of the subject when subscribing to it. Another option is to emit initial value as [] instead of null

Reply:
Fixed